lib/llog/api.go

```diff
@@ -0,0 +1,201 @@
+package llog
+
+import (
+	"io"
+)
+
+// Level indicates logging level. The higher the value, the less important.
+// Any non-negative integer is always less important than a named constant.
+type Level int
+
+const (
+	// Least important
+	Info Level = -1 - iota
+	Warn
+	// Most important
+	Err
+)
+
+// Set sets this level from string equivalent. s must be "Err",
+// "Warn", "Info", "0", "1", "2", ... or else Set returns an error. Set
+// ignores the case in s, so passing "ERR", "Err", or "err" as s has the same
+// effect.
+func (l *Level) Set(s string) error {
+	// TODO
+	return nil
+}
+
+// String returns the string representation of this level in all caps.
+func (l *Level) String() string {
+	// TODO
+	return ""
+}
+
+// FlushWriter allows the logging of fatal and panic messages to notify the
+// underlying writer that the process is terminating.
+type FlushWriter interface {
+	io.Writer
+	// Logging a fatal or panic message calls this just before terminating
+	// the process.
+	Flush() error
+}
+
+// A Sink synchronizes the writing of log messsges to a particular Writer.
+// A Sink instance is safe to use with multiple goroutines. A client will
+// create one Sink to control access to a single Writer and then create one
+// or more Logger instances, each with possibly different configurations,
+// to write to that same Sink.
+type Sink struct {
```

Comment:
I see this package as being very thin, just like the current log package from the Go standard library. It's essentially just a log formatting package. People can call the Write() method of the Writer concurrently too, so the Writer needs to deal with this regardless.